src/figuresClasses.ts

```diff
@@ -2,10 +1,98 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
```

Comment:
The `shape` field is being set in the constructor with a default value, which is unnecessary because the class `Triangle` should already encapsulate the idea of its shape being a triangle. You can remove the default value assignment from the constructor and declare the `shape` field in the class with a default value or as a readonly property.